tests/support/mod.rs

```diff
@@ -203,7 +203,7 @@ impl FunctionTransform for MockTransform {
                 v.push_str(&self.suffix);
                 log.insert(vector::config::log_schema().message_key(), Value::from(v));
             }
-            Event::Metric(metric) => match metric.data.value {
+            Event::Metric(metric) => match metric.value() {
                 MetricValue::Counter { ref mut value } => {
                     *value += self.increase;
                 }
```

Comment:
This test transform does direct manipulations of the metric data, which is exactly what we want to avoid doing. I'm not sure how to do this better. I could move it into core, but it doesn't really belong there. I could work around by cloning the value, manipulating it, and sticking it back in wholesale, but that's pretty ugly. It's not performance critical because it's a test. Any other thoughts? See the whole file: https://github.com/timberio/vector/blob/9412d19485f0cc1436b4fcdb7d19e6fefa2858b8/tests/support/mod.rs#L206

Reply:
Hmm, you know, now that the `Transform` trait is in core this might not be out of place there. That is, I've intended for there to be some source/transform/sink implementations in core that aren't exposed to end-users but are useful for internal performance testing work. No reason why the same can't be true for correctness tests. I wouldn't call this `MockTransform` if you move it, but I'd +1 a move into core test-only.